.github/workflows/ci.yml

```diff
@@ -1,45 +1,61 @@
 name: CI
 on:
   push:
-    branches: ["main"]
-    tags: ["v*"]
+    branches: [ "main" ]
   pull_request:
-    branches: ["*"]
+  release:
+    types: [ 'published' ]
 
-env: 
+env:
   JAVA_OPTS: -Xmx6G
 
 jobs:
   build:
     name: CI
     strategy:
       fail-fast: false
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-24.04
     steps:
       - uses: actions/checkout@v3
         with:
           fetch-depth: 0
 
-      - uses: coursier/cache-action@v6      
+      - name: Set Swap Space
+        uses: pierotofy/set-swap-space@master
+        with:
+          swap-size-gb: 10
```

Comment:
Is this some sort of universal recommendation? What does this help with?

Reply:
The CI fails otherwise. It seemed like insufficient memory, so I added it. And now it's green.